File: if_else.py
'''
Flow control - if else
        - evalueaza conditii si bifurca codul in functie de rezultat
'''

# #if
#
# print('pornim radio')
# piesa_faina=True
# if piesa_faina == True:
#     print('dam mai tare')
#     print('fredonam')
# print('oprim radio')
#
# #if else
#
# #daca nr este par printam acest lucru, altfel printam impar
# nr = 3
#
# # ce inseamna par? se imparte exact la 2
# # ce inseamna ca se imparte la 2? ne da restul 0
#
# if nr % 2 == 0:
#     print('nr par')
# else:
#     print('impar')
#
# # este un nr pozitiv?
# if (nr > 0):
#     print('pozitiv')
# else:
#     print('negativ')
#
# # if, else if, else
#
# # cum ne saluta robotelul in functie de ora?
# # luam date de la tastatura
# # ne asiguram ca sunt transformate din string in int
#
# ora = int(input('Introdu ora '))
# if ora < 0:
#     print('ora invalida')
# elif ora <= 11:
#     print('buna dimineata')
# elif ora <= 18:
#     print('buna ziua')
# elif ora <= 21:
#     print('buna seara')
# elif    ora <= 24:
#     print('noapte buna')
# else:
#     print('ora invalida')
    # CTRL + /  - pt a comenta/decomenta mai multe linii

# robotel telefonic (switch din JAVA)
opt = int(input('alege optiunea: '))
if opt == 0:
    print('meniu anterior')
elif opt == 1:
    print('ati ales RO')
elif opt == 2:
    print('ati ales EN')
else:
    print('nu am identificat optiunea. mai incearca')

File: src/ifElse.java
import java.util.Scanner;

/*Flow control - if else
        - evalueaza conditii si bifurca codul in functie de rezultat
*/
public class ifElse {
    public static void main(String[] args) {
//        System.out.println("pornim radio");
//        boolean piesa_faina = true; // invat calculatorul daca piesa este faina sau nu
//        //daca piesa e frumoasa, dau mai tare
//        if (piesa_faina == true) {
//            System.out.println("dau mai tare radio");
//            System.out.println("incep sa fredonez piesa");
//        }
//        System.out.println("oprim radio");
//
//        //if else
//
//        //daca nr este par printam acest lucru, altfel printam impar
//        int nr = 4;
//        if (nr % 2 == 0) {
//            System.out.println("nr par");
//        } else {
//            System.out.println("impar");
//        }
//
//        //este un nr pozitiv?
//        if (nr > 0) {
//            System.out.println("pozitiv");
//        } else {
//            System.out.println("negativ");
//        }
//
//        // if, else if, else
//
//        // cum ne saluta robotelul in functie de ora?
//        // luam date de la tastatura
//        Scanner sc = new Scanner(System.in); //scanner = este un tip de date ( la fel ca string, int, char)
//        System.out.println("Introdu ora: ");
//        int ora = sc.nextInt();
//        if (ora < 0) {
//            System.out.println("ora invalida");
//        } else if (ora <= 11) {
//            System.out.println("buna dimineata");
//        } else if (ora <= 18) {
//            System.out.println("buna ziua");
//        } else if (ora <= 21) {
//            System.out.println("buna seara");
//        } else if (ora <= 24) {
//            System.out.println("noapte buna");
//        } else {
//            System.out.println("ora invalida");
//        }
        // CTRL + /  - pt a comenta/decomenta mai multe linii

        //switch
        //se foloseste cand stim valorile posibile (valori fixe) (exemplu robotelul de la telefonia mobila)
        System.out.println("alege optiunea: ");
        Scanner sc = new Scanner(System.in);
        int opt = sc.nextInt();
        switch (opt) {
            case 0:
                System.out.println("meniu anterior");
                break;
            case 1:
                System.out.println("RO");
                break;
            case 2:
                System.out.println("ENG");
                break;
            default:
                System.out.println("opt invalida");
        }
    }
}
